python/tank/deploy/tank_commands/install.py

```diff
@@ -84,12 +85,13 @@ def run_interactive(self, log, args):
         :param args: command line args
         """
 
-        if "--preserve-yaml" in args:
```

Comment:
I really wish all that boiler plate code for checkin the --use-pyyaml flag and log.info was refactored for all commands that update the yaml.

Then the if constants.LEGACY_YAML_PARSER_FLAG block would become

legacy_yaml_parser = _should_use_legacy_yaml_parser(args)